modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileSetManager.java

```diff
@@ -82,13 +91,39 @@ protected DbaseFileReader openDbfReader(boolean indexed) throws IOException {
             return null;
         }
 
+        Charset charset = store.getCharset();
+
+        if (store.isTryCpgFile()
+                && shpFiles.get(CPG) != null
+                && (!shpFiles.isLocal() || (shpFiles.isLocal() && shpFiles.exists(CPG)))) {
+            try (BufferedReader br =
+                    new BufferedReader(
+                            new InputStreamReader(
+                                    shpFiles.getInputStream(CPG, this), "ISO-8859-1"))) {
```

Comment:
better to use  `java.nio.charset.StandardCharsets.ISO_8859_1` instead of a string